px.py

```diff
@@ -388,7 +411,7 @@ def do_socket(self, xheaders={}, destination=None):
         keepalive = False
         ua = False
         cmdstr = "%s %s %s\r\n" % (self.command, self.path, self.request_version)
-        self.proxy_socket.send(cmdstr.encode("utf-8"))
+        self.proxy_socket.sendall(cmdstr.encode("utf-8"))
         dprint(cmdstr.strip())
```

Comment:
Per docs, sendall() will raise an exception if everything wasn't sent. Do we need to handle that?

Reply:
If sendall() raises an exception something is wrong with the connection and for the most reasons send() would then raise an exception too. I don't see a great difference to the current behaviour. It would be good to handle the exceptions of either send() and sendall() but I think this has not to be done in this pull request.